### ISSUE TITLE
[use case: chat app and parallel regular MUA] let chat not disturb regular email clients

### ISSUE THREAD

**Issue by testbird** [deltachat/deltachat-core#60]
Tested: A (SIEVE) filter rule on the imap server that routes incoming chat messages directly into the chat folder prevents other email clients like thunderbird or K-9 from getting flooded with chat messages and return receipts (and mobile users getting notified not only by delta chat but also the email client).

However, delta chat is currently only watching the INBOX folder. Could you also make it watch the Chats folder in addition to the INBOX, in order to make it work with the filter rule?

Ideally delta chat itself could also upload the SIEVE filter definition to the imap server automatically.

wiki page:
https://github.com/deltachat/deltachat-core/wiki/IMAP-strategy

**Comment by testbird** [comment]
Watching the Chats folder may also enable quick syncing between delta chat clients on different devices.

**Comment by testbird** [comment]
The other use case: "one MUA for chat and email"

**Comment by sketet** [comment]
I find this behavior very frustrating.
I use one email account with alias addresses for the several uses (so one for the chats).
Normally I used a filter that move the mails (to chat address) to the chat folder, but DC don't notify the mail in chat folder.
And leaving the chat mails in the inbox I receive the notify by both the apps, in my case K9 and DC. I really hate this!

Is it possible that the chat mails don't pass through the inbox?

**Comment by testbird** [comment]
Subfolder notification seems possible in K-9 https://android.stackexchange.com/a/151463, so hopefully this can get fixed.

**Comment by sketet** [comment]
the problem is not K-9, it works fine.
according to me, the problem is DC that use the inbox, DC should move directly the mails to the chat folder so it does not create problems with other messages.

**Comment by r10s** [comment]
watching multiple folders is possible, but it is not smart, see https://github.com/deltachat/deltachat-core/issues/12 , currently, Delta synchonized non-INBOX-folders about every half hour, but does not notify for new messages found there. If we do so, we would probably also run into [multi-device issues](https://github.com/deltachat/deltachat-core/issues/52) as the mails in the Chats folder may also come from another device.

So, at all, there are lots of corner cases. If anyone sees a solution to make things _easier_ and not more complicated, this would be very welcome :)

**Comment by testbird** [comment]
>watching multiple folders is possible, but it is not smart, see #12 
I don't find that explained there.

So, is DC also reading the Spam folder every 30 minutes? That could be one cause for https://github.com/deltachat/deltachat-android/issues/239

One idea for a clear cut distinct separation of chat vs. email was to let DC only show unrelated messages (not a reply, start new chat), if the subject starts with "Chat: ". Something a DC client does by default and is easily understood.

**Comment by testbird** [comment]
> If anyone sees a solution to make things easier and not more complicated, this would be very welcome :)

I suspect that trying upload a SIEVE filter definition and watching only the inbox and chat folder by default could solve a couple of the basic issues and corner cases. The multi DC client problems could be worked out in second step and seen as an advanced use-case?

**Comment by testbird** [comment]
**inbox watch:** allows to see and move non DC initiated new chats and replies not starting with "Chat: " into the chat directory

**Comment by testbird** [comment]
Now I thought of it: Watching only the inbox and chat may not work with all setups that already move emails into sender specific folders. Scanning all other folders (except spam) may still be necessary, if the "Chat: " subject filter can not be inserted in a way that guarantees it is processed first to get the corner cases of replies without a "Chat" subject. (Is this just documentable (filter order, reply subject) and negligible?)

**Comment by sketet** [comment]
According to me and concerned my use (K-9 for the classic mail use and DC with an alias address for the chats) should be simple, it would be enough move the chat messages directly (and not after read the message) into the chat folder and consequently notifications for messages from the chat folder.
The moving, I think it's simple, already made this but immediately instead of after reading.
The notifications, it could be done, k-9 allow you to decide a class of folder to receive notification and Gmail app allow you to decide every folder for the notification.

**Comment by testbird** [comment]
Created a wiki page to avoid the serialization of updates:
 https://github.com/deltachat/deltachat-core/wiki/IMAP-strategy

**Comment by testbird** [comment]
> If we [watch the INBOX], we would probably also run into multi-device issues as the mails in the Chats folder may also come from another device.

https://github.com/deltachat/deltachat-core/issues/52 suggests to me that watching would not be different than what happens with current polling.

I revised the IMAP strategy wiki design doc and can't think of any corner case anymore.

If possible I would suggest to plan in like a monthly session, giving a break to working on new features, to fix usability issues, to keep up and enable the testing the early adopters are doing. Maybe your already doing this, it could make users more patient/understanding.

**Comment by csb0730** [comment]
If the title of this issue is in mind (!) this means too, that
1. messages which are not intended for chats must not be deleted from server.
2. messages which possibly are intended for another MUA (the main account handling one) may be moved but not deleted !

IMHO a simple global option "do not delete messages on server" solves this problem.

In the moment the user only has the possibility to "delete messages on server" or do nothing (cancel message delete at local device - nonsatisfying :-(  ).

**Comment by csb0730** [comment]
See [deltachat-android 321](https://github.com/deltachat/deltachat-android/issues/321) too

**Comment by testbird** [comment]
@csb0730 Again, the way the current "contact requests" work is unfortunate. The messages there should automatically disappear when the user is done with them in his classic email client (MUA). And the user should only need to look into the the deltachat "contact requests" (i.e. Unnapproved / Ignored Emails) to have a quick look into his emails. It allows him to start a chat with new contacts. There should be no need to delete emails there, as this is done in the MUA, but if there is an obvious spam message, for example, pressing the DELETE MESSAGE button in deltachat should actually also delete it from the server. Otherwise, the spam would have to be deleted again in the MUA, which is not resulting in a consistent user experience.

Generally, there is only the exception of users using POP clients to retrieve emails from the server (your case). That setup does not allow for a user experience that is consistently synced (through the IMAP server) across clients, instead it requires to manage the emails separately and repetitively on every client. The options you wish for in Only to support this. The options https://github.com/deltachat/deltachat-android/issues/321 https://github.com/deltachat/deltachat-android/issues/327 you wish for are only needed for this case, if the user insists to -- unnecessarily -- clean up the "contact requests" manually, instead of letting the emails disappear automatically when they are deleted on the server.

But unfortunately deltachat currently continues to keep the emails (contact requests) even after they were deleted on the server. That is the problem, and is the thing that needs to be fixed here to make DC usable.

**Comment by csb0730** [comment]
@testbird, I'm with You but not completely.

I'm using DC, to see what other mails are incoming, read them and to decide if I really need to start my standard MUA (as You described it). Often I start my standard MUA only after some days! Quite 95% of email I can handle with DC :-)

In spite of that I wish to clean local Contact Requests without danger. Why DC forces the user to delete at server too? If You make any failure and delete an important mail it is deleted forever. Therefore a really prefer to keep mails at server.  The old behaviour of DC (not to delete at server) or a user option for that would IMHO really be needed.
I would prefer a simple option for that. This would solve all my problems :-)

Anyway my mailserver deletes automatically mails which are older than a certain date (this is a setting at server and can be adjusted as needed).

Spam is not really an issue, because mail server does filtering it very efficient at his level and DC don't sees most of it anyway :-)

**Comment by testbird** [comment]
I think we're on the same page, except for one thing.

> Why DC forces the user to delete at server too?

You're related issues are simply open. I don't think complaining over and over, allover that frequently is helpful. Also, at least I acknowledged the necessity for two config options several times, and tried to explain the preferable default (syncing).

Still, a workaround or fix for you (combining IMAP and POP features, assuming you like a MUA maintaining a local copy), could be to have a MUA that can use IMAP to accesses the server (showing the remote folder tree) and supports filter rules to copy the messages from the IMAP account to the local mail storage (do what POP would do).

> I'm using DC, to see what other mails are incoming,

Incoming? Hey, that may be a another nice naming alternative for "contact requests",
**"(...) Incoming Emails"** ??? https://github.com/deltachat/deltachat-core/issues/48

> I wish to clean local Contact Requests without danger.

Yes, like everyone. There could be separate issue, if the messages are not properly moved into the trash by default. (Haven't checked.)

**Comment by csb0730** [comment]
@testbird, please don't understand me wrong. My intention is only to describe more clear the problem.

With my standard MUA I'm using imap (really ;-) ) but in spite of that I prefer to organize my mails at local device and not at server side. So mail server is for me only the flexible temporary distribution point.

Regarding replacement for "Contact Requests" naming: I didn't thought about a new proposal for that here but ... why not? :-)
The keyword _Email_ is important for You in that, right? Yes that's good :-)
"Incoming Emails" - Why not :+1:

**Comment by r10s** [comment]
targeted by https://github.com/deltachat/deltachat-core/milestone/2

closing this for now here, of course, the discussion can go on in the new support forum at https://support.delta.chat
